src/bin/bootleby.rs

```diff
@@ -112,7 +112,10 @@ fn check_for_override(
 
         // Safety: the exclusivity check above is more than enough to make
         // obtaining a reference to this variable _once_ safe:
-        unsafe { &mut TRANSIENT_OVERRIDE }
+        #[allow(static_mut_refs)]
+        unsafe {
+            &mut TRANSIENT_OVERRIDE
```

Comment:
Since we've already checked exclusivity above, the current Official Ralf Way of fixing this is

```suggestion
            &mut *&raw mut TRANSIENT_OVERRIDE
```

...or the equivalent with `addr_of_mut!` if Clippy gives you any guff. (It had a bug about this in 1.83 iirc.)

Reply:
I like this better and will add it!

Reply:
The compiler likes it, clippy frowns upon it
```
error: immediately dereferencing a reference
   --> src/bin/bootleby.rs:115:23
    |
115 |         unsafe { &mut *&raw mut TRANSIENT_OVERRIDE }
    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `TRANSIENT_OVERRIDE`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#deref_addrof
    = note: `-D clippy::deref-addrof` implied by `-D warnings`
    = help: to override `-D warnings` add `#[allow(clippy::deref_addrof)]`
    ```
    I think I'm going to keep the original skip for now